Sources/Account/Account+TransactionOperations.swift

```diff
@@ -281,5 +281,214 @@ extension Account {
                 }
             }
         }
+
+        func preparePresignedInputTransaction(
+            presignedInput: SignedContingentInput,
+            memoType: MemoType,
+            fee: Amount,
+            completion: @escaping (Result<PendingTransaction, TransactionPreparationError>)
+            -> Void
+        ) {
+            let amountToSend = presignedInput.requiredAmount
+            let amountToReceive = presignedInput.rewardAmount
+
+            let (unspentTxOuts, ledgerBlockCount) =
+                self.account.readSync {
+                    ($0.unspentTxOuts(tokenId: amountToSend.tokenId),
+                     $0.knowableBlockCount)
+                }
+
+            guard ledgerBlockCount <= presignedInput.tombstoneBlockIndex else {
+                serialQueue.async {
```

Comment:
fwiw I prefer this approach to guards over the one where its factored into a function in `Sources/Account/Account+SCIOperations.swift`

Reply:
✅